pkg/controller/azvolumeattachment.go

```diff
@@ -594,9 +634,16 @@ func (r *reconcileAzVolumeAttachment) triggerDetach(ctx context.Context, attachm
 		return err
 	}
 
-	if err := r.detachVolume(ctx, azVolumeAttachment.Spec.UnderlyingVolume, azVolumeAttachment.Spec.NodeName); err != nil {
-		klog.Errorf("failed to detach volume %s from node %s: %v", azVolumeAttachment.Spec.UnderlyingVolume, azVolumeAttachment.Spec.NodeName, err)
-		return err
+	// check if underlying volume attachment exists before proceeding with detachment
+	if azVolumeAttachment.Annotations != nil {
```

Comment:
Is this check necessary? My understanding is that triggerDetach will only be called if the VolumeAttachment is deleted. Can there be a case when triggerDetach is called while the VolumeAttachment still exists?

Reply:
This is essentially to prevent the detachment from happening when the CRI is being deleted from uninstallation of the driver.

Reply:
Sure, makes sense